src/ast/expr.rs

```diff
@@ -140,23 +169,60 @@ impl Expr {
 
     pub fn map(&self, f: &mut impl FnMut(&Self) -> Self) -> Self {
         match self {
-            Expr::Lit(_) => f(self),
-            Expr::Var(_) => f(self),
-            Expr::Call(op, children) => {
+            GenericExpr::Lit(..) => f(self),
+            GenericExpr::Var(..) => f(self),
+            GenericExpr::Call(ann, op, children) => {
                 let children = children.iter().map(|c| c.map(f)).collect();
-                f(&Expr::Call(*op, children))
+                f(&GenericExpr::Call(ann.clone(), op.clone(), children))
+            }
+        }
+    }
+
+    // TODO: cannon function may want to take the annotation of variables
```

Comment:
what does this comment mean?

Reply:
Currently, `subst_leaf` takes a leaf but not an annotation over the leaf, so it has to "make up" annotations for the returned `GenericExpr`. A better approach is for `subst_leaf` to also take the annotation, which we should implement after we use real non-`()` annotations

(also added the above explanation to the code)